lib/schema.js

```diff
@@ -302,34 +325,57 @@ class Metaschema {
     }
   }
 
-  // Create category instance
-  //   def - <Object> | <string> field definition
+  // Create domain instance
+  //   def - <Object> field definition
   //   value - <any> value to check/create
   // Returns: instance of `def`
-  createInstance(def, value) {
-    let name;
-    if (typeof def === 'string') name = def;
-    else name = def.category || def.domain;
-
-    const decorator = extractDecorator(def);
-    const category = this.categories.get(name);
-    if (category && category.factory) {
-      return category.factory(value);
-    }
+  createDomainInstance(def, value) {
+    if (def.domain === 'Id') return new Uint64(value);
+    if (def.domain === 'Money') return new Money(value);
+    const domain = this.domains.get(def.domain);
+    const decorator = extractDecorator(domain);
+    if (domain.normalize) value = domain.normalize(value, null, domain);
+    if (domain.parse) value = domain.parse(value);
     if (decorator === 'List') {
-      const cat = this.categories.get(def.category);
+      const cat = this.categories.get(domain.category);
```

Comment:
`domain.category` looks weird, you should probably return `null` in this case.

Reply:
Maybe it would be better to move it to `createInstance`? I guess `List` might be used like this:
```javascript
{
  field: List('SomeCategory') 
  // in definition it will look like List { category: 'SomeCategory' }
}
```